### PR TITLE
Emit to the pip-compile's header an original command

### LINKED ISSUE #726: Command in autogenerated requirements.txt can be shortened
When I run `pip-compile`, my requirements.txt has

```
#
# This file is autogenerated by pip-compile
# To update, run:
#
#    pip-compile --output-file requirements.txt requirements.in
#
```

But I think the `--output-file requirements.txt` can just be dropped (for brevity) when the written file itself is named `requirements.txt`.

I'm recommending this because `pip-compile` already goes ahead and modifies `requirements.txt` when no options are specified. Thoughts?

### PR DESCRIPTION
**Pull request by atugushev**
Uses sys.argv to emit an original command. Closes #726.

**Changelog-friendly one-liner**: Emit to the pip-compile's header an original command

##### Contributor checklist

- [X] Provided the tests for the changes.
- [X] Requested a review from another contributor.
- [X] Gave a clear one-line description in the PR (that the maintainers can add to CHANGELOG.md on release).
- [X] Assign the PR to an existing or new milestone for the target version (following [Semantic Versioning](https://blog.versioneye.com/2014/01/16/semantic-versioning/)).

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -383,12 +382,6 @@ def test_generate_hashes_with_editable():
             ],
         )
     expected = (
-        '#\n'
-        '# This file is autogenerated by pip-compile\n'
-        '# To update, run:\n'
-        '#\n'
-        '#    pip-compile --generate-hashes --output-file requirements.txt requirements.in\n'
-        '#\n'
```
Comment (blueyed) [inline tests/test_cli.py]: Couldn't this be left/adjusted?

Comment (atugushev) [reply tests/test_cli.py]: I think it's possible. On the other hand this header has already been tested in `test_write_header`. Do you think it is necessary to test it once again?

Comment (blueyed) [reply tests/test_cli.py]: Not really.

Just wondered first why it was removed - but that was not due to it not being there then anymore.

Comment (atugushev) [reply tests/test_cli.py]: Thank you for the feedback 👍

Comment (blueyed) [reply tests/test_cli.py]: You're welcome!  Thanks for working on pip-tools! :)

**Comment by blueyed** [comment]
Restarted Travis builds - there appears to have been a temporary failure with downloads for py34/py35/pypy/pypy3.

**Comment by atugushev** [comment]
@blueyed yes, thank you! It seems everything is okay now.

**Comment by blueyed** [comment]
Great.  Will let you / somebody else merge this.

**Comment by vphilippon** [comment]
`pip-tools v3.4.0` released

**Pull request merged**